ZOOKEEPER-3184: Use the same method to generate website as documentation

tamaashu opened the pull request:
With this change we use the same method to generate the website as we do for the documentation.
This way we would get rid of Jekyll and would not need to install anything besides maven on the build machine(s).

Also replaced the page structure to be able to use BootStrap for CSS, where
I've got "strong inspiration" from the Apache HBase website.

Change-Id: I94f79298f9ce8f97510ee71c65f857eaa41d59ed

eolivelli commented:
Awesome work

May we rename 'hbase' files to 'zookeeper' ?

Is bootstrap compatible from a licensing perspective? Should we add some 'NOTICE' ?

tamaashu commented:
> Awesome work
> 
> May we rename 'hbase' files to 'zookeeper' ?

Even removed them. I've updated to the newest BootStrap, where I've dropped everything I copied from HBase.
 
> Is bootstrap compatible from a licensing perspective? Should we add some 'NOTICE' ?

BootStrap needs Popper.js and jQuery. All the three have MIT License, which is compatible with Apache 2.0.

https://getbootstrap.com/docs/4.1/about/license/
https://github.com/FezVrasta/popper.js/blob/master/LICENSE.md
https://jquery.org/license/

I'll check how to make a NOTICE file.

tamaashu commented:
Hi @eolivelli, could you please check it again? Since I changed it fundamentally, I think it needs a new review.

@anmolnar, @hanm could you please review these changes?

Thanks.

anmolnar commented:
@tamaashu Have you changed the design of the website intentionally?

tamaashu commented:
@anmolnar yes. This design is responsive, adopts to the screen size, simple, easy to read, but still modern. It's close to HBase's design, but since I've used 2 version newer bootstrap it differs in a few things. It contains all the data as before with an additional page for events, which is an Apache requirement afaik.

tamaashu commented on src/main/resources/markdown/bylaws.md:
```diff
@@ -1,7 +1,19 @@
-title: Project Bylaws
----
+<!--
+Copyright 2002-2004 The Apache Software Foundation
```
@eolivelli here you can download a tarball: https://drive.google.com/open?id=1GubCh20SPSS6rCi71syA9H9YCILR_D1t

tamaashu replied:
2002-2004 can be probably dropped. I'll remove them.

nkalmar commented:
I also like the design, compared to HBase, I can definitely see the inspiration :)

Good job, +1 non-binding

lvfangmin commented on README.md:
```diff
@@ -1,49 +1,24 @@
-1. `jekyll build` in this directory.
-2. `cp -RP _released_docs _site/doc` - this will include the documentation (see "sub-dir" section below) in the generated site.
+1. `mvn clean install` in this directory.
+2. `cp -RP _released_docs _target/html` - this will include the documentation (see "sub-dir" section below) in the generated site.
```
After 'mvn clean install' there is no _target dir but only target, is this a typo or am I missing any step?

tamaashu replied:
Thanks for noting it, it was a typo. Fixed.

tamaashu replied:
Hi @lvfangmin, thanks for the detailed review.

Yes it is the right page. The documentation links point into the void, since the documentations are generated in a separate process and are manually copied into their final place. Once the generated page is copied into its place the links won't be broken.

The wiki page must be updated after the content will be changed. I've already got my account, so am able to edit it, just wait for the commit of this change.

lvfangmin replied:
Thanks @tamaashu, can you update the readme to point out how to copy those doc? 

One suggest, the top and bottom layout is a bit strange, they're too wide comparing to the main content on that page, can we make those not that wide?

tamaashu commented:
Hi @hanm,

> * The layout is a little bit odd - can we remove the empty spaces on both left and right side of the web page? Use HBase main page for a reference: https://hbase.apache.org/.

The layout is quite the same as on https://www.apache.org/ (e.g. https://www.apache.org/licenses/)

> * All links under 'Documentation' dropdown menu is broken.

The documentation is generated in a separate process and has to be copied manually into its place, this is why the links look broken.

anmolnar commented:
I'm still happy with the changes. Confirmed that we don't need any special approval for website changes. @hanm @lvfangmin any objections?

anmolnar commented:
@lvfangmin How about now?

anmolnar commented:
Committed to `website` branch. Thanks @tamaashu !
